src/main/java/appeng/client/render/tesr/CableBusFastTESR.java

```diff
@@ -0,0 +1,306 @@
+			{
+				if( facades.containsKey( EnumFacing.UP ) )
+				{
+					max.y -= thickness;
+				}
+
+				if( facades.containsKey( EnumFacing.DOWN ) )
+				{
+					min.y += thickness;
+				}
+
+				if( facades.containsKey( EnumFacing.SOUTH ) )
+				{
+					max.z -= thickness;
+				}
+
+				if( facades.containsKey( EnumFacing.NORTH ) )
+				{
+					min.z += thickness;
+				}
+			}
+
+			builder.addCube( min.x, min.y, min.z, max.x, max.y, max.z );
+		}
+		else
+		{
+			Vector3f busMin = new Vector3f( (float) busBounds.minX * 16, (float) busBounds.minY * 16, (float) busBounds.minZ * 16 );
+			Vector3f busMax = new Vector3f( (float) busBounds.maxX * 16, (float) busBounds.maxY * 16, (float) busBounds.maxZ * 16 );
+
+			if( side == EnumFacing.UP || side == EnumFacing.DOWN )
+			{
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMax.z, 16.0f, 16.0f, 16.0f );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, 0.0f, 16.0f, 16.0f, busMin.z );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMin.z, busMin.x, 16.0f, busMax.z );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, busMax.x, 0.0f, busMin.z, 16.0f, 16.0f, busMax.z );
+			}
+			else if( side == EnumFacing.NORTH || side == EnumFacing.SOUTH )
+			{
+				if( facades.get( EnumFacing.UP ) != null )
+				{
+					max.y -= thickness;
+				}
+
+				if( facades.get( EnumFacing.DOWN ) != null )
+				{
+					min.y += thickness;
+				}
+
+				this.renderSegmentBlockCurrentBounds( builder, min, max, busMax.x, 0.0f, 0.0f, 16.0f, 16.0f, 16.0f );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, 0.0f, busMin.x, 16.0f, 16.0f );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, busMin.x, 0.0f, 0.0f, busMax.x, busMin.y, 16.0f );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, busMin.x, busMax.y, 0.0f, busMax.x, 16.0f, 16.0f );
+			}
+			else
+			{
+				if( facades.get( EnumFacing.UP ) != null )
+				{
+					max.y -= thickness;
+				}
+
+				if( facades.get( EnumFacing.DOWN ) != null )
+				{
+					min.y += thickness;
+				}
+
+				if( facades.get( EnumFacing.SOUTH ) != null )
+				{
+					max.z -= thickness;
+				}
+
+				if( facades.get( EnumFacing.NORTH ) != null )
+				{
+					min.z += thickness;
+				}
+
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMax.z, 16.0f, 16.0f, 16.0f );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, 0.0f, 16.0f, 16.0f, busMin.z );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, 0.0f, busMin.z, 16.0f, busMin.y, busMax.z );
+				this.renderSegmentBlockCurrentBounds( builder, min, max, 0.0f, busMax.y, busMin.z, 16.0f, 16.0f, busMax.z );
+			}
+		}
+		return ret;
+	}
+
+	private void renderSegmentBlockCurrentBounds( CubeBuilder builder, Vector3f min, Vector3f max, float minX, float minY, float minZ, float maxX, float maxY, float maxZ )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00107)